File: Domain/Scheduler.cs

namespace HostTool.Domain
{
    public class Scheduler
    {
        public Guid SchedulerId { get; set; }
        public string SchedulerName { get; set; } = string.Empty;
        public string SchedulerDescription { get; set; }
        public string SchedulerPath { get; set; }
        public bool Active { get; set; }
        public bool RunAll { get; set; }
    }

    public class SchedulerDay
    {
        public Guid SchedulerDayId { get; set; }
        public Guid SchedulerId { get; set; }
        public int Day { get; set; }
        public string From { get; set; }
        public string To { get; set; }
        public bool Active { get; set; }
    }

    public class SchedulerDTO
    {
        public Scheduler? master { get; set; }
        public  List<SchedulerDay>? detail { get; set; }

        //0 --- edit ---1 xóa
        public int EditMode { get; set; } = 0;

    }
}

File: view/src/api/scheduler.ts
// src/services/schedulerService.ts
import http from './http'

export interface Scheduler {
  schedulerId: string // Guid => string
  schedulerName: string
  schedulerDescription: string
  schedulerPath: string
  active: boolean
  runAll: boolean
}

export interface SchedulerDay {
  schedulerDayId: string // Guid => string
  schedulerId: string // Guid => string
  day: number
  from: string // .NET string giữ nguyên
  to: string // .NET string giữ nguyên
  active: boolean
}

export interface CreateUpdateSchedulerDto {
  master?: Scheduler // nullable trong C# => optional trong TS
  detail?: SchedulerDay[] // List<T> => T[]
  editMode: number // int => number
}


export const schedulerService = {
  create(input: CreateUpdateSchedulerDto) {
    return http.post<boolean>('/tool', input)
  },

  update(id: string, input: CreateUpdateSchedulerDto) {
    return http.put<boolean>(`/tool`, input)
  },

  get() {
    return http.get<Scheduler[]>(`/tool/GetSchedulers`)
  },

  getDetail(id: string) {
    return http.get<SchedulerDay[]>(`/tool/GetSchedulerDays/${id}`)
  },

  getList() {
    return http.get<CreateUpdateSchedulerDto[]>('/tool/GetListScheduler', {})
  },
}
